pycode/memilio-surrogatemodel/memilio/surrogatemodel/GNN/data_generation_nodamp.py

```diff
@@ -0,0 +1,328 @@
+        "./data/contacts/baseline_work.txt")
+    baseline_contact_matrix3 = os.path.join(
+        "./data/contacts/baseline_other.txt")
+
+    baseline = np.loadtxt(baseline_contact_matrix0) \
+        + np.loadtxt(baseline_contact_matrix1) + \
+        np.loadtxt(baseline_contact_matrix2) + \
+        np.loadtxt(baseline_contact_matrix3)
+
+    return baseline
+
+def generate_data(
+        num_runs, path, input_width, days, save_data=True):
+    """! Generate dataset by calling run_secir_simulation (num_runs)-often
+   @param num_runs Number of times, the function run_secir_simulation is called.
+   @param path Path, where the datasets are stored.
+   @param input_width number of time steps used for model input.
+   @param label_width number of time steps (days) used as model output/label.  
+   @param save_data Option to deactivate the save of the dataset. Per default true.
+   """
+    
+    population = get_population()
+    days_sum = days + input_width - 1
+
+    data = {"inputs": [],
+                "labels": [],
+                }
+
+    # show progess in terminal for longer runs
+    # Due to the random structure, theres currently no need to shuffle the data
+    bar = Bar('Number of Runs done', max=num_runs)
+
+    for _ in range(num_runs):
+
+            data_run = run_secir_groups_simulation(
+                days_sum, population)
+
+            inputs = np.asarray(data_run).transpose(1, 2, 0)[: input_width]
+            data["inputs"].append(inputs)
+
+            data["labels"].append(np.asarray(data_run).transpose(1, 2, 0)[input_width:])
+
+            bar.next()
+
+    bar.finish()
+
+    if save_data:
+            num_groups = int(np.asarray(data['inputs']).shape[2] / 8)
+            transformer = FunctionTransformer(np.log1p, validate=True)
+
+            # Scale inputs
+            inputs = np.asarray(
+                data['inputs']).transpose(2, 0, 1, 3).reshape(num_groups * 8, -1)
+            scaled_inputs = transformer.transform(inputs)
+            original_shape_input = np.asarray(data['inputs']).shape
+            
+            # Step 1: Reverse the reshape
+            reshaped_back = scaled_inputs.reshape(original_shape_input[2], 
+                                                  original_shape_input[0], 
+                                                  original_shape_input[1], 
+                                                  original_shape_input[3])
+
+            # Step 2: Reverse the transpose
+            original_inputs = reshaped_back.transpose(1, 2, 0, 3)
+            scaled_inputs = original_inputs.transpose(0, 3, 1, 2)
+
+            
+            # Scale labels
+            labels = np.asarray(
+                data['labels']).transpose(2, 0, 1, 3).reshape(num_groups * 8, -1)
+            scaled_labels = transformer.transform(labels)
+            original_shape_labels = np.asarray(data['labels']).shape
+            
+            # Step 1: Reverse the reshape
```

Comment:
Step 1 and 2 of what? That's not quite clear here? i mean the genereal intention